src/JuMPArray.jl

```diff
@@ -5,67 +5,76 @@
 
 # This code is unused for now. See issue #192
 
-immutable JuMPArray{T,N,R<:OrdinalRange} <: JuMPContainer
+immutable JuMPArray{T,N,NT<:NTuple} <: JuMPContainer{T}
     innerArray::Array{T,N}
-    name::Symbol
-    indexsets::NTuple{N,R}
+    indexsets::NT
+    lookup::NTuple{N,Dict}
+    meta::Dict{Symbol,Any}
 end
 
-function Base.getindex{T}(d::JuMPArray{T,1,UnitRange{Int}},index::Real)
-    @inbounds return d.innerArray[index - start(d.indexsets[1])+1]
+function JuMPArray{T,N}(innerArray::Array{T,N}, indexsets::NTuple{N})
+    JuMPArray(innerArray, indexsets, ntuple(N) do i
```

Comment:
Does type inference work here? The mix of anonymous functions might make it tricky.